langchain4j-core/pom.xml

```diff
@@ -173,6 +173,45 @@
             </build>
         </profile>
 
+        <profile>
+            <id>compliance</id>
+            <activation>
+                <property>
+                    <name>compliance</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.honton.chas</groupId>
+                        <artifactId>license-maven-plugin</artifactId>
+                        <version>0.0.3</version>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>compliance</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <!-- We don't check test related dependencies -->
```

Comment:
After giving it a second though I start to feel like we should stick to permissive licenses for all dependencies, including test.

Our friend ChatGPT says:

> The fact that you're using a non-permissively licensed library for testing might not directly affect the distribution of your Apache2 licensed project, as the testing code and its dependencies might not be included in the production distribution. However, if the testing code is distributed with the rest of the project (which is often the case in open-source projects), you may be obligated to comply with the non-permissive license's conditions for that code.
> 
> In the case of a license like the GPL, for example, these conditions may include making the source code of your whole project available under the GPL as well, even if only part of the code is using the GPL'ed library. This is because the GPL is a "copyleft" license that is designed to ensure that derivative works also remain open source.

Reply:
What do you think @Shastick ?

Reply:
fair point, updated